modules/lwjgl/core/src/main/java/org/lwjgl/system/ThreadLocalUtil.java

```diff
@@ -184,21 +193,27 @@ public static void setCapabilities(long capabilities) {
 
     // Ensures FUNCTION_MISSING_ABORT will be called even if no context is current,
     public static void setFunctionMissingAddresses(int functionCount) {
-        // OpenJDK: NULL
-        // GraalVM Native Image: pointer to UnimplementedWithJNIEnvArgument function (see #875)
-        long RESERVED0_NULL = memGetAddress(JNI_NATIVE_INTERFACE);
-
         long ptr = JNI_NATIVE_INTERFACE + CAPABILITIES_OFFSET;
 
+        // aka. reserved3
         long currentTable = memGetAddress(ptr);
+
         if (functionCount == 0) {
-            if (currentTable != RESERVED0_NULL) {
+            if (currentTable == FUNCTION_MISSING_ABORT) {
                 FUNCTION_MISSING_ABORT_TABLE = NULL;
                 getAllocator().free(currentTable);
-                memPutAddress(ptr, NULL);
+                memPutAddress(ptr, RESERVED3_NULL);
             }
         } else {
-            if (currentTable != RESERVED0_NULL) {
+            // OpenJDK: NULL
+            // EspressoVM: pointer to VM internal
+            // GraalVM Native Image: pointer to UnimplementedWithJNIEnvArgument function (see #875)
+            long RESERVED0_NULL = memGetAddress(JNI_NATIVE_INTERFACE);
+
+            boolean slotAvailable = currentTable == RESERVED3_NULL; // on HotSpot or Espresso
```

Comment:
I think it's better to check against `NULL` here, that allows detecting VMs or environment which put something meaningful in `reserved3`.